backend/models/brains.py

```diff
@@ -50,130 +50,41 @@ def create(cls, *args, **kwargs):
         )  # pyright: ignore reportPrivateUsage=none
 
     def get_user_brains(self, user_id):
-        response = (
-            self.commons["supabase"]
-            .from_("brains_users")
-            .select("id:brain_id, brains (id: brain_id, name)")
-            .filter("user_id", "eq", user_id)
-            .execute()
-        )
-        return [item["brains"] for item in response.data]
+        return self.commons["db"].get_user_brains(user_id)
```

Comment:
We follow the use of commons here, but I am not sure pushing the db object in a dict enables the VSCode python server to be helpful and know the type of the variable?
We should use the typing of the `Repository` in another MR.

Reply:
@gozineb actually I noticed that for the other items on `commons`, my VSCode does not seem to be able to infer a type, even though we have `CommonsDep = Annotated[dict, Depends(common_dependencies)]` below. WDYT of having an explicit Dataclass `CommonsDep` instead of resolving the types ad-hoc?

Reply:
Agree https://github.com/StanGirard/quivr/pull/453/commits/44c6a2ac93f1f53b3917ae187a9c82141c13d71f

Reply:
@Noezor really like it, super clean !

Reply:
Keeping this for another PR, wdyt about the current state of this one?

Reply:
There are conflicts that needs to be fixed before merging @Noezor ;)